### PR TITLE
feat(focus-mvp-android-device): Device Orientation Handler

### PR DESCRIPTION
**Pull request by brocktaylor7**
#### Details

This adds the DeviceOrientationHandler which allows for classes to subscribe to orientation change events.

##### Motivation

Feature Work

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Added/updated relevant unit test(s)
- [x] Ran `./gradlew fastpass` from `AccessibilityInsightsForAndroidService`
- [x] PR title _AND_ final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`).

### CODE REVIEW

File: AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/AccessibilityInsightsForAndroidService.java [hunk 1]
```diff
@@ -140,6 +143,16 @@ public void onAccessibilityEvent(AccessibilityEvent event) {
     }
   }
 
+  @Override
+  public void onConfigurationChanged(Configuration newConfig) {
+    super.onConfigurationChanged(newConfig);
+    if(newConfig.orientation == this.deviceOrientationHandler.getOrientation()){
```
Comment (waabid) [inline AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/AccessibilityInsightsForAndroidService.java:149]: Feels like this is already being handled inside the orientation handler already so this shouldn't be necessary here.

(I'd prefer to be checking in the orientation handler anyways)

File: AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    this.orientation = orientation;
+    this.onOrientationChangedListeners = new ArrayList<>();
+  }
+
+  public void subscribe(Consumer<Integer> listener) {
+    this.onOrientationChangedListeners.add(listener);
+  }
+
+  public void setOrientation(int orientation) {
+    if (this.orientation != orientation) {
```
Comment (waabid) [inline AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java:23]: nit: can flip this around by using guard clause here.

File: AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      this.orientation = orientation;
+      this.emitChanged(orientation);
+    }
+  }
+
+  public int getOrientation() {
```
Comment (waabid) [inline AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java:29]: Can likely get rid of this as it won't be necessary for the moment.

File: AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/AccessibilityInsightsForAndroidService.java (hunk 1 above)
Comment (brocktaylor7) [reply AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/AccessibilityInsightsForAndroidService.java]: This is a good call, I've made this change.

File: AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java (hunk 2 above)
Comment (brocktaylor7) [reply AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java]: done!

File: AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java (hunk 3 above)
Comment (brocktaylor7) [reply AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java]: removed this, updated the tests!

**Review by waabid — APPROVED** [review]

**Pull request merged**